### PR TITLE
ROX-23068: Add cluster list to Platform CVE detail page

### PR DESCRIPTION
**Pull request by dvail**
## Description

Title.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Visit Platform CVEs, and click the link of a CVE to view the affected clusters list.

Loading:
image

Error:
image

Empty (should never occur here though...):
image

Filtered Empty:
image

Success:
image

Test clicking the cluster link:
image

Test clicking the breadcrumb link in the header:
image

### CODE REVIEW

File: ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/NodeCvePage.tsx [hunk 1]
```diff
@@ -79,7 +79,9 @@ function NodeCvePage() {
             <PageTitle title={`Node CVEs - NodeCVE ${nodeCveName}`} />
             <PageSection variant="light" className="pf-v5-u-py-md">
                 <Breadcrumb>
-                    <BreadcrumbItemLink to={workloadCveOverviewCvePath}>CVEs</BreadcrumbItemLink>
+                    <BreadcrumbItemLink to={workloadCveOverviewCvePath}>
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/NodeCves/NodeCve/NodeCvePage.tsx:82]: Is `workloadCveOverviewCvePath` correct for Node CVEs?

File: ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/utils/stringUtils.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { ClusterType } from 'types/cluster.proto';
+import { ensureExhaustive } from 'utils/type.utils';
+
+export function displayClusterType(type: ClusterType): string {
+    switch (type) {
+        case 'GENERIC_CLUSTER':
+            return 'Generic';
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/utils/stringUtils.ts:7]: You might double check with backend coworkers whether `'GENERIC_CLUSTER'` is possible.

For comparison:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Clusters/cluster.helpers.tsx#L37-L41

It sticks in my mind the absence of that constant value came up, and we learned it was for Docker Swarm clusters in the early days of the product. When in doubt, better safe than sorry, though.

File: ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/PlatformCve/PlatformCvePage.tsx [hunk 3]
```diff
@@ -36,12 +59,21 @@ function PlatformCvePage() {
...
+        searchFilter: querySearchFilter,
+    });
+
     return (
         <>
             <PageTitle title={`Platform CVEs - PlatformCVE ${cveName}`} />
             <PageSection variant="light" className="pf-v5-u-py-md">
                 <Breadcrumb>
-                    <BreadcrumbItemLink to={workloadCveOverviewCvePath}>CVEs</BreadcrumbItemLink>
+                    <BreadcrumbItemLink to={workloadCveOverviewCvePath}>
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/PlatformCve/PlatformCvePage.tsx:74]: Similar question whether `workloadCveOverviewCvePath` is correct for Platform CVEs?

File: ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/PlatformCve/AffectedClustersTable.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                <Tr>
+                    <Th>Cluster</Th>
+                    <Th>Cluster type</Th>
+                    <Th>Kubernetes version</Th>
+                </Tr>
+            </Thead>
+            <TbodyUnified
+                tableState={tableState}
+                colSpan={3}
+                emptyProps={{ message: 'No clusters have been reported for this CVE' }}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/PlatformCve/AffectedClustersTable.tsx:61]: You can rewrite as you suggested for a similar sentence about nodes.

**Review by pedrottimark — APPROVED** [review]
Clear to read and straightforward to understand. A few comments, as usual.

File: ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/utils/stringUtils.ts (hunk 2 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/utils/stringUtils.ts:7]: Agreed that I think this is effectively deprecated at this point. It is still included in the API documentation though, so I figured I would cover it for completeness.

**Pull request merged**